elfi/examples/gauss.py

```diff
@@ -1,30 +1,81 @@
-"""An example implementation of a Gaussian noise model."""
+"""Example implementations of Gaussian noise models."""
 
 from functools import partial
 
 import numpy as np
 import scipy.stats as ss
 
 import elfi
+from elfi.examples.gnk import euclidean_multidim
 
 
-def Gauss(mu, sigma, n_obs=50, batch_size=1, random_state=None):
-    """Sample the Gaussian distribution.
+def gauss(mu, sigma, n_obs=50, batch_size=1, random_state=None):
+    """Sample the 1-D Gaussian distribution.
 
     Parameters
     ----------
     mu : float, array_like
     sigma : float, array_like
     n_obs : int, optional
     batch_size : int, optional
-    random_state : RandomState, optional
+    random_state : np.random.RandomState, optional
+
+    Returns
+    -------
+    y_obs : array_like
+        1-D observation.
 
     """
-    # Standardising the parameter's format.
-    mu = np.asanyarray(mu).reshape((-1, 1))
-    sigma = np.asanyarray(sigma).reshape((-1, 1))
-    y = ss.norm.rvs(loc=mu, scale=sigma, size=(batch_size, n_obs), random_state=random_state)
-    return y
+    # Handling batching.
+    batches_mu = np.asanyarray(mu).reshape((-1, 1))
+    batches_sigma = np.asanyarray(sigma).reshape((-1, 1))
+
+    # Sampling observations.
+    y_obs = ss.norm.rvs(loc=batches_mu, scale=batches_sigma,
+                        size=(batch_size, n_obs), random_state=random_state)
+    return y_obs
+
+
+def gauss_nd_mean(*mu, cov_matrix, n_obs=15, batch_size=1, random_state=None):
+    """Sample an n-D Gaussian distribution.
+
+    Parameters
+    ----------
+    *mu : array_like
+        Mean parameters.
+    cov_matrix : array_like
+        Covariance matrix.
+    n_obs : int, optional
+    batch_size : int, optional
+    random_state : np.random.RandomState, optional
+
+    Returns
+    -------
+    y_obs : array_like
+        n-D observation.
+
+    """
+    n_dim = len(mu)
+
+    # Handling batching.
+    batches_mu = np.zeros(shape=(batch_size, n_dim))
+    for idx_dim, param_mu in enumerate(mu):
+        batches_mu[:, idx_dim] = param_mu
+    batches_cov = np.zeros(shape=(batch_size, n_dim, n_dim))
+    for idx_batch in range(batch_size):
+        batches_cov[idx_batch, :, :] = cov_matrix
```

Comment:
I don't think we need `batches_cov` at all, since it's the same for all batches now... I.e. just give `cov_matrix` as is to `ss.multivariate_normal.rvs`?